example/src/screens/Room.tsx

```diff
@@ -57,33 +58,53 @@ export const Room = () => {
     }, [])
   );
 
-  const videoViewWidth = (width - 3 * OFFSET_PER_ROW) / 2;
-  const smallScreenVideoWidth =
-    (height - HEADER_AND_FOOTER_SIZE - OFFSET_PER_ROW * (rowNum + 2)) / rowNum;
+  const isScreensharingTrack = (track: Membrane.Track) => {
+    return track.metadata.type === 'screensharing';
+  };
+
+  const isTrackFocused = (p: Participant) => {
+    return (
+      p.participant.id === focusedParticipantData?.participant.id &&
+      p.trackId === focusedParticipantData?.trackId
+    );
+  };
+
+  useEffect(() => {
+    const screencast = participants
+      .reverse()
+      .find((p) => p.tracks.some((t) => isScreensharingTrack(t)));
+
+    if (screencast) {
+      setFocusedParticipantData({
+        participant: screencast,
+        trackId: screencast.tracks.find((t) => isScreensharingTrack(t))!.id,
+      });
+    } else {
+      setFocusedParticipantData(null);
+    }
+  }, [
+    participants.filter((p) => p.tracks.some((t) => isScreensharingTrack(t)))
+      .length,
+  ]);
+
+  useEffect(() => {
+    const curretStateOfFocusedParticipant = participants.find(
+      (p) => p.id === focusedParticipantData?.participant.id
+    );
+    if (
+      focusedParticipantData?.participant !== curretStateOfFocusedParticipant
+    ) {
+      setFocusedParticipantData({
+        participant: curretStateOfFocusedParticipant!,
```

Comment:
what if pinned participant exits the call? I've got a crash

Reply:
Good catch, thanks. I haven't realised that the participant that is being focused could disappear.

Reply:
and get rid of this ! now, it's not necessary
and it caused the crash, I'm considering adding a lint rule to prohibit using ! :D

Reply:
Removed.